internal/autoupdate/autoupdate_sync.go

```diff
@@ -0,0 +1,222 @@
+package autoupdate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/go-github/v35/github"
+	"go.uber.org/zap"
+
+	"github.com/simplesurance/goordinator/internal/logfields"
+)
+
+type syncAction int
+
+const (
+	undefined syncAction = iota
+	enqueue
+	dequeue
+)
+
+// Sync synchronized the states of the autoupdater queues with the current
+// pull-request state at GitHub.
+// Pull-Request information are queried from github.
+// If a PR meets a condition to be enqueued for auto-updates it is enqueued.
+// If it meets a condition for not being autoupdated, it is dequeued.
```

Comment:
```suggestion
// If it meets the conditions for not being autoupdated, it is dequeued.
```

Reply:
there are multiple conditions, so the current version should be correct, do you agree?

Reply:
yes :+1: